base/regex.jl

```diff
@@ -61,8 +61,21 @@ Regex(pattern::AbstractString) = Regex(pattern, DEFAULT_COMPILER_OPTS, DEFAULT_M
 
 function compile(regex::Regex)
     if regex.regex == C_NULL
```

Comment:
Someday we should provide a helper for this pattern (https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_once.html, https://en.cppreference.com/w/cpp/thread/call_once, http://docs.libuv.org/en/v1.x/threading.html#once-only-initialization)